packages/clerk-js/src/ui/components/SubscriptionDetails/index.tsx

```diff
@@ -0,0 +1,664 @@
+        <SubscriberTypeContext.Provider value={props.for}>
+          <SubscriptionDetailsInternal {...props} />
+        </SubscriberTypeContext.Provider>
+      </SubscriptionDetailsContext.Provider>
+    </Drawer.Content>
+  );
+};
+
+type UseGuessableSubscriptionResult<Or extends 'throw' | undefined = undefined> = Or extends 'throw'
+  ? {
+      upcomingSubscription?: CommerceSubscriptionResource;
+      activeSubscription: CommerceSubscriptionResource;
+      anySubscription: CommerceSubscriptionResource;
+      isLoading: boolean;
+    }
+  : {
+      upcomingSubscription?: CommerceSubscriptionResource;
+      activeSubscription?: CommerceSubscriptionResource;
+      anySubscription?: CommerceSubscriptionResource;
+      isLoading: boolean;
+    };
+
+function useGuessableSubscription<Or extends 'throw' | undefined = undefined>(options?: {
+  or?: Or;
+}): UseGuessableSubscriptionResult<Or> {
+  const { data: subscriptions, isLoading } = useSubscriptions();
+  const activeSubscription = subscriptions?.find(sub => sub.status === 'active');
+  const upcomingSubscription = subscriptions?.find(sub => sub.status === 'upcoming');
+
+  if (options?.or === 'throw' && !activeSubscription) {
+    throw new Error('No active subscription found');
+  }
+
+  return {
+    upcomingSubscription,
+    activeSubscription: activeSubscription as any, // Type is correct due to the throw above
+    anySubscription: (upcomingSubscription || activeSubscription) as any,
+    isLoading,
+  };
+}
+
+const SubscriptionDetailsInternal = (props: __internal_SubscriptionDetailsProps) => {
+  const { organization: _organization } = useOrganization();
+  const [subscriptionForCancellation, setSubscriptionForCancellation] = useState<CommerceSubscriptionResource | null>(
+    null,
+  );
+  const [confirmationOpen, setConfirmationOpen] = useState(false);
+
+  const {
+    buttonPropsForPlan: _buttonPropsForPlan,
+    isDefaultPlanImplicitlyActiveOrUpcoming: _isDefaultPlanImplicitlyActiveOrUpcoming,
+  } = usePlansContext();
+
+  const { data: subscriptions, isLoading } = useSubscriptions();
+  const { activeSubscription } = useGuessableSubscription();
+
+  if (isLoading) {
+    return (
+      <Spinner
+        sx={{
+          margin: 'auto',
+        }}
+      />
+    );
+  }
+
+  if (!activeSubscription) {
+    // Should never happen, since Free will always be active
+    return null;
+  }
+
+  return (
+    <SubscriptionForCancellationContext.Provider
+      value={{
+        subscription: subscriptionForCancellation,
+        setSubscription: setSubscriptionForCancellation,
+        confirmationOpen,
+        setConfirmationOpen,
+      }}
+    >
+      <Drawer.Header title={localizationKeys('commerce.subscriptionDetails.title')} />
+
+      <Drawer.Body>
+        <Col
+          gap={4}
+          sx={t => ({
+            padding: t.space.$4,
+            overflowY: 'auto',
+          })}
+        >
```

Comment:
Maybe expose sx on the Drawer.Body to remove the need to nest a Col inside it for padding in this usage. Or add the padding to the SubscriptionCard.